src/index.ts

```diff
@@ -25,50 +23,32 @@ client.on('guildCreate', guild => {
     console.log(
         `New guild joined: ${guild.name} (id: ${guild.id}). This guild has ${guild.memberCount} members!`
     )
-    client.user?.setActivity(`Serving ${client.guilds.cache.size} servers`)
+    client.user?.setActivity(`with yarn`)
 })
 
 client.on('guildDelete', guild => {
     // this event triggers when the bot is removed from a guild.
     console.log(`I have been removed from: ${guild.name} (id: ${guild.id})`)
-    client.user?.setActivity(`Serving ${client.guilds.cache.size} servers`)
+    client.user?.setActivity(`with yarn`)
 })
 
-client.on('message', async (message: Discord.Message) => {
+client.on('message', async (message: Message) => {
     // This event will run on every single message received, from any channel or DM.
 
-    // It's good practice to ignore other bots. This also makes your bot ignore itself
-    // and not get into a spam loop (we call that "botception").
+    // Ignore other bots
     if (message.author.bot) return
 
-    // Also good practice to ignore any message that does not start with our prefix,
-    // which is set in the configuration file.
-
+    // On messages without prefix run these commands
     if (message.content.indexOf(config.prefix) !== 0) {
         // Randomly meow when a message is received
-        if (Math.random() < 0.05) {
-            await message.channel.send('Meow.')
-        }
+        randomMeow(message)
         return
     }
-    // Here we separate our "command" name, and our "arguments" for the command.
-    // e.g. if we have the message "+say Is this the real life?" , we'll get the following:
-    // command = say
-    // args = ["Is", "this", "the", "real", "life?"]
-    const args = message.content
-        .slice(config.prefix.length)
-        .trim()
-        .split(/ +/g)
-    const command = args.shift()!.toLowerCase()
 
-    if (command === 'ping') {
-        // Calculates ping between sending a message and editing it, giving a nice round-trip latency.
-        const m = await message.channel.send('Ping?')
-        m.edit(
-            `Pong! Latency is ${m.createdTimestamp -
-                message.createdTimestamp}ms.`
-        )
-    }
+    const [command, args] = parseMessage(message)
```

Comment:
this could all be tidied  determineCommand could just take in the message  object ajd just handle the parse message stuff, and even execute it